src/analyses/ai.h

```diff
@@ -218,18 +221,16 @@ class ai_baset
     const goto_programt &goto_program,
     const irep_idt &identifier) const;
 
+  // the work-queue is sorted using the history's less operator
```

Comment:
Aha, in which case it would be highly advisable for history's operator< to order primarily by location number for efficiency?

Reply:
Yes when the histories are in the same function.  Originally I implemented the histories and the de-recursion in the same patch but no-one would ever want to review that so I separated them.  Some details like this are important but are subject to rapid change.